Compiled from "CWE396_Catch_Generic_Exception__Exception_15.java"
public class testcases.CWE396_Catch_Generic_Exception.CWE396_Catch_Generic_Exception__Exception_15 extends testcasesupport.AbstractTestCase {
  public testcases.CWE396_Catch_Generic_Exception.CWE396_Catch_Generic_Exception__Exception_15();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method testcasesupport/AbstractTestCase."<init>":()V
       4: return

  public void bad() throws java.lang.Throwable;
    Code:
       0: bipush        7
       2: lookupswitch  { // 1
                     7: 20
               default: 37
          }
      20: ldc           #7                  // String Test
      22: invokestatic  #9                  // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      25: pop
      26: goto          42
      29: astore_1
      30: ldc           #17                 // String Caught Exception
      32: invokestatic  #19                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      35: aload_1
      36: athrow
      37: ldc           #25                 // String Benign, fixed string
      39: invokestatic  #19                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      42: return
    Exception table:
       from    to  target type
          20    26    29   Class java/lang/Exception

  private void good1() throws java.lang.Throwable;
    Code:
       0: bipush        8
       2: lookupswitch  { // 1
                     7: 20
               default: 28
          }
      20: ldc           #25                 // String Benign, fixed string
      22: invokestatic  #19                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      25: goto          45
      28: ldc           #7                  // String Test
      30: invokestatic  #9                  // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      33: pop
      34: goto          45
      37: astore_1
      38: ldc           #17                 // String Caught Exception
      40: invokestatic  #19                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      43: aload_1
      44: athrow
      45: return
    Exception table:
       from    to  target type
          28    34    37   Class java/lang/NumberFormatException

  private void good2() throws java.lang.Throwable;
    Code:
       0: bipush        7
       2: lookupswitch  { // 1
                     7: 20
               default: 37
          }
      20: ldc           #7                  // String Test
      22: invokestatic  #9                  // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      25: pop
      26: goto          42
      29: astore_1
      30: ldc           #17                 // String Caught Exception
      32: invokestatic  #19                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      35: aload_1
      36: athrow
      37: ldc           #25                 // String Benign, fixed string
      39: invokestatic  #19                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      42: return
    Exception table:
       from    to  target type
          20    26    29   Class java/lang/NumberFormatException

  public void good() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: invokevirtual #29                 // Method good1:()V
       4: aload_0
       5: invokevirtual #34                 // Method good2:()V
       8: return

  public static void main(java.lang.String[]) throws java.lang.ClassNotFoundException, java.lang.InstantiationException, java.lang.IllegalAccessException;
    Code:
       0: aload_0
       1: invokestatic  #37                 // Method mainFromParent:([Ljava/lang/String;)V
       4: return
}
